source/MaterialXGenShader/ShaderGenerator.h

```diff
@@ -92,7 +92,7 @@ class MX_GENSHADER_API ShaderGenerator
     virtual void emitFunctionDefinitions(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const;
 
     /// Add the function call for a single node.
-    virtual void emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage, bool checkScope = true) const;
+    virtual void emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const;
```

Comment:
This parameter has no effect since the feature is not in use. But note that this is a change to shader generation API. 

Since there is a default value on the parameter calls to the API are not effected, and it's only derived shader generators that my be affected by this change, if they happen to override this function. I suspect that's very rare. But let me know if this API breakage needs to be handled with more care, and we should keep the parameter in for now.

Reply:
@niklasharrysson The approach I've been taking for backwards compatibility is to add a deprecated wrapper for the original function interface, so that existing clients can continue to build their code with the latest version of MaterialX.

Now that we've set C++14 as the baseline for MaterialX, we can additionally mark these wrappers with the `[[deprecated]]` keyword, so that clients receive a warning at compile time.  Here's an example of this pattern in the current codebase:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/7fa59fa0088c930e2a8c6cdc4b0181c9678d07e6/source/MaterialXGenShader/Util.h#L93